rs/system_api/src/lib.rs

```diff
@@ -696,6 +702,12 @@ struct MemoryUsage {
     /// The current amount of execution memory that the canister is using.
     current_usage: NumBytes,
```

Comment:
Very good question. That was what I believed it to be, but in this case, it is not. I still do not have an answer why that is the case. But the initial implementation added only the `stable_memory_usage` field and `wasm_memory_usage` was calculated as `current_usage` - `stable_memory_usage`. That created the problem that `wasm_memory_usage`, without a memory limit, was greater(a couple of bytes) than 4GiB in some tests which should not happen.  So changing the way of calculating the wasm memory fixed the problem.

Reply:
I'm not sure I fully understand this (there used to be no limit on `wasm_memory_usage`, so it would sometimes exceed 4 GB?).

But regardless, looking at the code, it still looks to me like `current_usage == stable_memory_usage + wasm_memory_usage` is something of an implicit invariant. Whether that's the case or not, the three fields seem tightly coupled in some way (as opposed to e.g. `stable_memory_usage`, `wasm_memory_usage` and `current_message_usage`, which are totally independent of one another) and it makes sense to document that.

Reply:
AFAIK, when no limit is set, it is supposed that wasm32 is used to apply limitation on the size of Wasm memory to 4 GiB (32 bits for addressing). 

Regarding the relation between the fields I agree that it should be documented, but I could not find more information about what `current_usage` should represent, and from the code I also infer that it should be `current_usage == stable_memory_usage + wasm_memory_usage`, but it was not the case. I will create follow-up ticket for this https://dfinity.atlassian.net/browse/EXC-1733.

Reply:
Thank you for creating a follow-up ticket.